src/sources/core.js

```diff
@@ -128,20 +135,62 @@ class CoreDataset extends Dataset {
 
     const prefix = prefixParts.join("/");
 
-    if (doesPathExist(prefix)) {
-      return this;
-    }
+    if (!doesPathExist(prefix)) {
+      /* the path doesn't match any available datasets exactly */
+      const nextDefaultPart = global.availableDatasets.defaults[sourceName][prefix];
 
-    /* if we are here, then the path doesn't match any available datasets exactly */
-    const nextDefaultPart = global.availableDatasets.defaults[sourceName][prefix];
+      if (nextDefaultPart) {
+        const dataset = new this.constructor(this.source, [...prefixParts, nextDefaultPart], this.versionDescriptor);
+        return dataset.resolve();
+      }
+    }
 
-    if (nextDefaultPart) {
-      const dataset = new this.constructor(this.source, [...prefixParts, nextDefaultPart]);
-      return dataset.resolve();
+    /**
+     * If we have a versionDescriptor (extracted from the requesting URL) then:
+     * (1) if the format is not YYYY-MM-DD we redirect to a URL without a version descriptor
+     * (2) if it is different from the identified versionId, then redirect to the latter
+     * (3) if it's the same as the identified versionId then fallthrough
+     */
+    if (this.versionDescriptor && this.versionDescriptor!==LATEST) {
+      if (!this.versionDescriptor.match(/^\d{4}-\d{2}-\d{2}$/)) {
+        console.log("\tresolve -> remove version descriptor")
+        const dataset = new this.constructor(this.source, [...prefixParts], LATEST);
+        return dataset.resolve();
+      }
+      if (this.versionDescriptor!==this.versionId) {
+        console.log("\tresolve -> ", this.versionId)
+        const dataset = new this.constructor(this.source, [...prefixParts], this.versionId);
+        return dataset.resolve();
+      }
     }
 
     return this;
   }
+
+  /**
+   * Parse a human-readable versionDescriptor (currently only in YYYY-MM-DD format)
+   * and return the closest available versionID (YYYY-MM-DD string) and the associated
+   * versionUrls (an object linking available file types to their access URLs)
+   * @param {string} versionDescriptor 
+   * @returns [versionId, versionUrls]
+   */
+  versionInfo(versionDescriptor) {
+    if (versionDescriptor===LATEST) return [LATEST, undefined];
+    const versions = new ResourceVersions(this.source.name, 'dataset', this.pathParts.join("/"));
+    const versionDate = versions.closestDate(versionDescriptor);
+    if (!versionDate || versionDate===LATEST) return [LATEST, undefined];
+    return [versionDate, versions.subresourceUrls(versionDate)];
+  }
+
+}
+
+class CoreNarrative extends Narrative {
+  versionInfo(versionDescriptor) {
+    if (versionDescriptor!==LATEST) {
+      utils.verbose(`versions not implemented for ${this.source.name} narratives (version descriptor: "${versionDescriptor}")`)
+    }
+    return [LATEST, undefined];
+  }
```

Comment:
> instead of making the stub the default implementation in Resource?

I think this is the approach I've now taken, no? (Perhaps this comment was made before that change?)

Reply:
(Yep, it was made against an earlier version of the code, as it was a continuation of review there.)